### DISCUSSION TITLE
How to reload pdf even if the document(file) is not changed

### DISCUSSION THREAD

**Discussion by kobecow** [wojtekmaj/react-pdf#1202] [Q&A]
I'm trying to implement a feature to show rectangles on PDF. Thanks to this [discussion](https://github.com/wojtekmaj/react-pdf/issues/979), I could implement it with canvasRef and useRef.

Now I want to implement a feature of checkbox to turning on/off showing rectangles.
I tried to use useState for rectangles.(e.g.  To show --> [rect1,rect2]  , not to show --> []).
A component containing Document and Page was called, but onRenderSuccess of Page was not called and previous rectangles remained.

How can I reload PDF with the same file ?

**Comment by kobecow** [comment]
Let me simplify this question. In this example, I want to put a button or checkbox, and then a user click it, then "Acme Inc" disappear on PDF.
https://codesandbox.io/s/react-pdf-watermark-5wylx?file=/src/index.jsx

**Comment by wojtekmaj** [comment]
The easiest way to do this would be to take advantage of `key` React prop to force Page component to re-render, like so:

https://codesandbox.io/s/react-pdf-watermark-toggleable-890w1i
